src/dav.js

```diff
@@ -0,0 +1,376 @@
+const { createClient } = require('webdav')
+const parser = require('./xmlParser.js')
+
+export class Dav {
+  constructor (baseUrl, baseUrlv2) {
+    this._XML_CHAR_MAP = {
+      '<': '&lt;',
+      '>': '&gt;',
+      '&': '&amp;',
+      '"': '&quot;',
+      "'": '&apos;'
+    }
+
+    this.baseUrl = baseUrl
+
+    this.userName = null
+
+    this.password = null
+
+    this.xmlNamespaces = {
+      'DAV:': 'd',
+      'http://owncloud.org/ns': 'oc'
+    }
+
+    this.xmlNamespacesComponents = {
+      d: 'DAV:',
+      oc: 'http://owncloud.org/ns'
+    }
+
+    this.client = createClient(baseUrl, {})
+    this.clientv2 = createClient(baseUrlv2, {})
+  }
+
+  _escapeXml (s) {
+    return s.replace(/[<>&"']/g, function (ch) {
+      return this._XML_CHAR_MAP[ch]
+    })
+  };
+
+  /**
+   * Generates a propFind request.
+   *
+   * @param {string} url Url to do the propfind request on
+   * @param {Array} properties List of properties to retrieve.
+   * @param {string} depth "0", "1" or "infinity"
+   * @param {Object} [headers] headers
+   * @return {Promise}
+   */
+  propFind (path, properties, depth, headers, options = {}) {
+    if (typeof depth === 'undefined') {
+      depth = '0'
```

Comment:
why not having the depth & headers parameter having default values?  
`propFind (path, properties, depth = '0', headers = {}, options = {}) {`